src/compile/axis/assemble.ts

```diff
@@ -21,7 +21,7 @@ export function assembleAxis(
     header: boolean // whether this is called via a header
   } = {header: false}
 ): VgAxis {
-  const {orient, scale, title, ...axis} = axisCmpt.combine();
+  const {orient, scale, title, zindex, ...axis} = axisCmpt.combine();
```

Comment:
How do we ensure that we don't forget anything here in the future?

Reply:
If you look at the code below, that's probably the only thing left that's not specific to grid / main axis.